### PR TITLE
[15.0][IMP] sale_invoice_plan: miscellaneous improvements 

### PR DESCRIPTION
**Pull request by yankinmax**
Improvements here are taken from `purchase_invoice_plan` and adapted to `sale.order` model.
Related PR's:
 - [ ] https://github.com/OCA/purchase-workflow/pull/1180
 - [ ] https://github.com/OCA/purchase-workflow/pull/1243
 - [ ] https://github.com/OCA/purchase-workflow/pull/1462
 - [ ] https://github.com/OCA/purchase-workflow/pull/1466

### CODE REVIEW

File: sale_invoice_plan/tests/test_sale_invoice_plan.py [hunk 1]
```diff
@@ -336,3 +284,56 @@ def test_03_unlink_invoice_plan(self):
...
+                ],
+            }
+        )
+        # Overall amount changed to 11000, install amount not changed, only percent changed.
+        self.assertEqual(self.so_service_so_edit.amount_total, 11000)
+        self.so_service.invoice_plan_ids._compute_amount()
+        self.assertEqual(first_install.amount, 1000)
+        # Why is it 9.090909 here??? What is wrong for SO
+        # AssertionError: 10.0 != 9.090909
+        self.assertEqual(first_install.percent, 9.090909)
```
Comment (yankinmax) [inline sale_invoice_plan/tests/test_sale_invoice_plan.py:339]: hello @kittiu, can you pls help to resolve the test issue? I'm not an expert in this module.

**Comment by yankinmax** [comment]
@kittiu can you pls take a look?

File: sale_invoice_plan/models/sale.py [hunk 2]
```diff
@@ -25,6 +25,21 @@ class SaleOrder(models.Model):
         compute="_compute_ip_invoice_plan",
         help="At least one invoice plan line pending to create invoice",
     )
+    ip_total_percent = fields.Float(
```
Comment (rousseldenis) [inline sale_invoice_plan/models/sale.py]: I'd prefer having significant names than shortcuts (ip)

Same for all entries

File: sale_invoice_plan/data/sale_data.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<odoo noupdate="1">
+    <record id="decimal_price" model="decimal.precision" forcecreate="True">
+        <field name="name">Purchase Invoice Plan Percent</field>
```
Comment (kittiu) [inline sale_invoice_plan/data/sale_data.xml]:
```suggestion
        <field name="name">Sales Invoice Plan Percent</field>
```

File: sale_invoice_plan/models/sale.py [hunk 4]
```diff
@@ -60,7 +83,7 @@ def create_invoice_plan(
         self.invoice_plan_ids.unlink()
         invoice_plans = []
         Decimal = self.env["decimal.precision"]
-        prec = Decimal.precision_get("Product Unit of Measure")
+        prec = Decimal.precision_get("Purchase Invoice Plan Percent")
```
Comment (kittiu) [inline sale_invoice_plan/models/sale.py]:
```suggestion
        prec = Decimal.precision_get("Sales Invoice Plan Percent")
```

File: sale_invoice_plan/models/sale_invoice_plan.py [hunk 5]
```diff
@@ -42,9 +42,15 @@ class SaleInvoicePlan(models.Model):
         help="Last installment will create invoice use remaining amount",
     )
     percent = fields.Float(
-        digits="Product Unit of Measure",
+        digits="Purchase Invoice Plan Percent",
```
Comment (kittiu) [inline sale_invoice_plan/models/sale_invoice_plan.py]:
```suggestion
        digits="Sales Invoice Plan Percent",
```

**Comment by kittiu** [comment]
- [x] https://github.com/OCA/purchase-workflow/pull/1180
- [x] https://github.com/OCA/purchase-workflow/pull/1243
- [x] https://github.com/OCA/purchase-workflow/pull/1462
- [x] https://github.com/OCA/purchase-workflow/pull/1466

File: sale_invoice_plan/models/sale_invoice_plan.py [hunk 6]
```diff
@@ -53,6 +59,11 @@ class SaleInvoicePlan(models.Model):
         string="Invoices",
         readonly=True,
     )
+    amount_invoiced = fields.Float(
+        compute="_compute_invoiced",
+        store=True,
+        readonly=False,
```
Comment (kittiu) [inline sale_invoice_plan/models/sale_invoice_plan.py]:
```suggestion
```

Comment (kittiu) [reply sale_invoice_plan/models/sale_invoice_plan.py]: Should be a readonly compute field.

**Comment by kittiu** [comment]
LGTM, just small comments.
Thanks for migrating these :+1:

**Review by kittiu — APPROVED** [review]

**Comment by yankinmax** [comment]
@rousseldenis @kittiu PR is updated with your suggestions, thanks for the reviews.

**Review by cesar-tecnativa — APPROVED** [review]

**Comment by kittiu** [comment]
@yankinmax can you resolve conflict?

**Comment by yankinmax** [comment]
@kittiu probably we can merge it as we have approves :smiley:

**Comment by kittiu** [comment]
Yep, but i don't have right to merge. Let's wait for @OCA/crm-sales-marketing-maintainers

**Comment by pedrobaeza** [comment]
What is the meaning of having such separate commits?

**Comment by yankinmax** [comment]
The idea was to have similar commits that were merged to `purchase_invoice_plan` as features were taken from there.
I can squash if we don't need it.

**Comment by pedrobaeza** [comment]
Are they adding a special recognition having them separated?

**Comment by yankinmax** [comment]
I've just named them from related merged PR's.

**Comment by yankinmax** [comment]
> Are they adding a special recognition having them separated?

Nope

**Comment by pedrobaeza** [comment]
Please squash it then and I merge.

**Comment by yankinmax** [comment]
@pedrobaeza can it be merged?

**Pull request merged**